src/main/java/be/syntra/devshop/DevshopBack/services/SearchServiceImpl.java

```diff
@@ -0,0 +1,85 @@
+package be.syntra.devshop.DevshopBack.services;
+
+import be.syntra.devshop.DevshopBack.entities.Product;
+import be.syntra.devshop.DevshopBack.models.SearchModel;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import java.util.Comparator;
+import java.util.List;
+import java.util.stream.Collectors;
+
+@Slf4j
+@Service
+public class SearchServiceImpl implements SearchService {
+    private SearchModel searchModel = null;
+    private ProductService productService;
+
+    @Autowired
+    public SearchServiceImpl(ProductService productService) {
+        this.productService = productService;
+    }
+
+    @Override
+    public void setSearchModel(SearchModel searchModel) {
+        this.searchModel = searchModel;
```

Comment:
It doesn't seem necessary or desirable to store SearchModel state in a service bean. Can you please elaborate on why this is done?

Reply:
because of ignorance i'm afraid, this will be much easier to remove/fix with **DEV-035-add-pagination-to-backend-services-and-repo**

Reply:
And after looking at it again i know what i did wrong, fixed now.